ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB/Impl/ChainSel.hs

```diff
@@ -499,15 +521,24 @@ chainSelectionForBlock cdb@CDB{..} blockCache hdr punish = do
         return tipPoint
 
       -- The block @b@ fits onto the end of our current chain
-      | pointHash tipPoint == headerPrevHash hdr -> do
+      | pointHash tipPoint == headerPrevHash hdr
+      , let newBlockFrag = curChain AF.:> hdr
+        -- TODO need to promptly reprocess this block if the LoE advances?
+      , Just maxExtra <- computeLoEMaxExtra cdbLoELimit loeFrag newBlockFrag -> do
```

Comment:
This necessarily one greater than `computeLoEMaxExtra cdbLoELimit loeFrag curChain`. Since this is the common case, maybe we could/should optimize via that observation. Could be used in the next case too.

Reply:
> This is necessarily one greater than `computeLoEMaxExtra cdbLoELimit loeFrag curChain`.

You probably mean "one *smaller*" (the longer the last argument of `computeLoEMaxExtra`, the smaller its return value)? In particular, when `computeLoEMaxExtra cdbLoELimit loeFrag curChain` is `Just 0` then `computeLoEMaxExtra cdbLoELimit loeFrag newBlockFrag` is `Nothing`.

In any case, I will leave a generic comment calling out potential (micro-?)optimizations as is already the case for the next `computeLoEMaxExtra` :+1: